internal/services/storage/storage_management_policy_resource.go

```diff
@@ -222,22 +225,22 @@ func resourceStorageManagementPolicy() *pluginsdk.Resource {
 
 func resourceStorageManagementPolicyCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Storage.ManagementPoliciesClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
 	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	storageAccountId := d.Get("storage_account_id").(string)
```

Comment:
I think we could omit this intermediate variable.

Reply:
GH is messing with me, this comment is supposed to refer to [line 232](https://github.com/hashicorp/terraform-provider-azurerm/blob/8ac01c6e9d5de3dc04c3642e984f5d435f095133/internal/services/storage/storage_management_policy_resource.go#L232)

Reply:
turns out subscriptionId can also go too anyway!